expression/testdata/partition_pruner_out.json

```diff
@@ -177,5 +177,757 @@
+        "Result": [
+          "1 1 1"
+        ],
+        "Plan": [
+          "TableReader_7 0.02 root partition:p0 data:Selection_6",
+          "└─Selection_6 0.02 cop[tikv]  or(and(eq(test_partition.t1.id, 1), eq(test_partition.t1.a, 1)), and(eq(test_partition.t1.a, 1), eq(test_partition.t1.id, 1)))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t2 where (id = 1 and a = 1 and b = 1) or (a = 1 and id = 1 and b=1)",
+        "Result": [
+          "1 1 1"
+        ],
+        "Plan": [
+          "TableReader_7 0.00 root partition:p0 data:Selection_6",
+          "└─Selection_6 0.00 cop[tikv]  or(and(and(eq(test_partition.t2.id, 1), eq(test_partition.t2.a, 1)), eq(test_partition.t2.b, 1)), and(and(eq(test_partition.t2.a, 1), eq(test_partition.t2.id, 1)), eq(test_partition.t2.b, 1)))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t3 where (id = 1 and a = 1) or (a = 1 and id = 1)",
+        "Result": [
+          "1 1 1"
+        ],
+        "Plan": [
+          "TableReader_7 0.02 root partition:p0 data:Selection_6",
+          "└─Selection_6 0.02 cop[tikv]  or(and(eq(test_partition.t3.id, 1), eq(test_partition.t3.a, 1)), and(eq(test_partition.t3.a, 1), eq(test_partition.t3.id, 1)))",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t3 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t1 where a is null",
+        "Result": [
+          "<nil> <nil> <nil>"
+        ],
+        "Plan": [
+          "TableReader_7 10.00 root partition:p1 data:Selection_6",
+          "└─Selection_6 10.00 cop[tikv]  isnull(test_partition.t1.a)",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t1 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t2 where a is null and b is null",
+        "Result": [
+          "<nil> <nil> <nil>"
+        ],
+        "Plan": [
+          "TableReader_7 0.01 root partition:p1 data:Selection_6",
+          "└─Selection_6 0.01 cop[tikv]  isnull(test_partition.t2.a), isnull(test_partition.t2.b)",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t2 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t3 where a is null",
+        "Result": [
+          "<nil> <nil> <nil>"
+        ],
+        "Plan": [
+          "TableReader_7 10.00 root partition:p1 data:Selection_6",
+          "└─Selection_6 10.00 cop[tikv]  isnull(test_partition.t3.a)",
+          "  └─TableFullScan_5 10000.00 cop[tikv] table:t3 keep order:false, stats:pseudo"
+        ]
+      },
+      {
+        "SQL": "select * from t1 where a=id and id >10",
+        "Result": null,
+        "Plan": [
+          "TableReader_7 2666.67 root partition:all data:Selection_6",
```

Comment:
It seems that we do not propagate `a > 10` from this condition when pruning. This can be optimized later.

Reply:
Yes, currently only support prune from equivalence condition.